packages/esm-form-engine-app/src/form-renderer/form-renderer.component.tsx

```diff
@@ -28,6 +28,13 @@ const FormRenderer: React.FC<FormRendererProps> = ({
   const { t } = useTranslation();
   const { schema, error, isLoading } = useFormSchema(formUuid);
 
+  const handleCloseWorkspace = useCallback(() => {
+    closeWorkspace({
+      ignoreChanges: true,
+      onWorkspaceClose: () => launchPatientWorkspace('clinical-forms-workspace'),
```

Comment:
I understand why this seems to fix the issue that we created ourselves. The side effect of this is a rerender of [this component](https://github.com/openmrs/openmrs-esm-patient-chart/blob/main/packages/esm-patient-forms-app/src/forms/form-entry.component.tsx#L75) which resets the props driving the visibility of the slot where the Form root component is rendered. With the evolution of the workspace, [this workspace registration](https://github.com/openmrs/openmrs-esm-patient-chart/blob/6d0b682a14a2265d38d3544a9d54f0e5672d9ae7/packages/esm-form-engine-app/src/index.ts#L16-L24) seems to be unnecessary and is the root cause of this issue. @CynthiaKamau try removing it and validate whether it solves the issue. 

The startupApp func should simply:

```ts

export function startupApp() {
  defineConfigSchema(moduleName, configSchema);
  // remove the workspace registration below
  // registerWorkspace({
  //   name: 'patient-form-entry-workspace',
  //   title: 'Clinical Form',
  //   load: getAsyncLifecycle(() => import('./form-renderer/form-renderer.component'), options),
  //   canMaximize: true,
  //   canHide: true,
  //   width: 'wider',
  //   type: 'clinical-form',
  // });
}
```

Reply:
> I understand why this seems to fix the issue that we created ourselves. The side effect of this is a rerender of [this component](https://github.com/openmrs/openmrs-esm-patient-chart/blob/main/packages/esm-patient-forms-app/src/forms/form-entry.component.tsx#L75) which resets the props driving the visibility of the slot where the Form root component is rendered. With the evolution of the workspace, [this workspace registration](https://github.com/openmrs/openmrs-esm-patient-chart/blob/6d0b682a14a2265d38d3544a9d54f0e5672d9ae7/packages/esm-form-engine-app/src/index.ts#L16-L24) seems to be unnecessary and is the root cause of this issue. @CynthiaKamau try removing it and validate whether it solves the issue.
> 
> The startupApp func should simply:
> 
> ```ts
> export function startupApp() {
>   defineConfigSchema(moduleName, configSchema);
>   // remove the workspace registration below
>   // registerWorkspace({
>   //   name: 'patient-form-entry-workspace',
>   //   title: 'Clinical Form',
>   //   load: getAsyncLifecycle(() => import('./form-renderer/form-renderer.component'), options),
>   //   canMaximize: true,
>   //   canHide: true,
>   //   width: 'wider',
>   //   type: 'clinical-form',
>   // });
> }
> ```

It does solve the issue , because the props were always undefined, Thanks